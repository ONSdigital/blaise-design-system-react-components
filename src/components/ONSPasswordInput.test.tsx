import React from "react";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { ONSPasswordInput } from "./ONSPasswordInput";

describe("ONS Password Input Test", () => {
    const Props = {};

    const labelProps = {
        label: "Submit",
    };

    const changeProps = {
        onChange: jest.fn(),
    };

    const undefinedChangeProps = {
        onChange: undefined,
    };

    function wrapper(renderFunc: any, props: any) {
        return renderFunc(
            <ONSPasswordInput
                value={props.value}
                label={props.label}
                placeholder={props.placeholder}
                marginTop={props.marginTop}
                onChange={props.onChange}
            />,
        );
    }

    it("matches Snapshot", () => {
        expect(wrapper(render, Props)).toMatchSnapshot();
    });

    it("should render with the correct label", () => {
        wrapper(render, labelProps);
        expect(screen.getByLabelText(labelProps.label)).toBeVisible();
    });

    it("should trigger callback on change, e.g. user types 'testing'", async () => {
        wrapper(render, changeProps);

        const passwordInput = screen.getByTestId("login-password-input");
        await userEvent.type(passwordInput, "testing");

        expect(changeProps.onChange).toHaveBeenCalledTimes(7);
    });

    it("should handle a click on the checkbox", async () => {
        wrapper(render, undefinedChangeProps);

        const toggleShowPassword = screen.getByLabelText("Show password");
        await userEvent.click(toggleShowPassword);

        expect(toggleShowPassword).toBeChecked();
    });
});
