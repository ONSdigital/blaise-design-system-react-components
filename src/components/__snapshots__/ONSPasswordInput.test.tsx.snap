// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ONS Password Input Test matches Snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <p
        class="ons-field"
      >
        <label
          class="ons-label"
          for="password"
        />
        <span
          class="ons-checkbox ons-checkbox--toggle"
        >
          <input
            autocomplete="new-password"
            class="ons-checkbox__input"
            id="password-toggle"
            name="show-password"
            type="checkbox"
          />
          <label
            class="ons-checkbox__label "
            for="password-toggle"
            id="password-toggle-label"
          >
            Show password
          </label>
        </span>
        <input
          class="ons-input ons-input--text ons-input-type__input ons-u-mt-xs"
          data-testid="login-password-input"
          id="password"
          type="password"
          value=""
        />
      </p>
    </div>
  </body>,
  "container": <div>
    <p
      class="ons-field"
    >
      <label
        class="ons-label"
        for="password"
      />
      <span
        class="ons-checkbox ons-checkbox--toggle"
      >
        <input
          autocomplete="new-password"
          class="ons-checkbox__input"
          id="password-toggle"
          name="show-password"
          type="checkbox"
        />
        <label
          class="ons-checkbox__label "
          for="password-toggle"
          id="password-toggle-label"
        >
          Show password
        </label>
      </span>
      <input
        class="ons-input ons-input--text ons-input-type__input ons-u-mt-xs"
        data-testid="login-password-input"
        id="password"
        type="password"
        value=""
      />
    </p>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
